runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -262,10 +263,18 @@ class ExecutionContext final {
     std::unique_ptr<MemoryManager> memory_man_ptr{nullptr};
     std::unique_ptr<PythonInterpreterGuard> py_guard{nullptr};
 
+    // PRNG
+    std::string seed;
+    std::mt19937 gen;
+
   public:
-    explicit ExecutionContext() : initial_tape_recorder_status(false)
+    explicit ExecutionContext(std::string seed = "") : initial_tape_recorder_status(false)
     {
         memory_man_ptr = std::make_unique<MemoryManager>();
+
+        this->seed = seed;
+        std::seed_seq seed_evolution(seed.begin(), seed.end());
+        this->gen = std::mt19937(seed_evolution);
```

Comment:
Just thought I'd leave this here because I was curious about how the initialization worked from an arbitrary string:
https://www.pcg-random.org/posts/cpp-seeding-surprises.html

(Nothing to change though)